### PR TITLE
Create Release pipeline for base-* images

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Added a pipeline to release base-* image

Accepts:
* target - Target base image, one of: base-server, base-builder, base-ci-builder, base-admin-tools
* commit - Commit sha to build from
* tag - New tag for the image

## Why?

Automating release process

## Checklist

1. Closes 

2. How was this tested:

executed the script using `nektos/act`
```
[Release Base-* Docker image/build-push-images]   ❓  ::endgroup::
[Release Base-* Docker image/build-push-images]   ✅  Success - Build-Push Server image
```

in progress

3. Any docs updates needed?

### CODE REVIEW

File: .github/workflows/release-base-image.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Metatags for the image
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: temporalio/${{ github.event.inputs.target }}
+
+      - name: Build-Push Server image
+        uses: docker/build-push-action@v2
+        with:
+          push: false
```
Comment (feedmeapples) [inline .github/workflows/release-base-image.yml:45]: need to change to true once tested

File: .github/workflows/release-base-image.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: Release Base-* Docker image
+
+on:
+  workflow_dispatch:
+    inputs:
+      target:
+        description: "Target base image, one of: base-server, base-builder, base-ci-builder, base-admin-tools"
```
Comment (jbreiding) [inline .github/workflows/release-base-image.yml:7]: Maybe this should be of type `choice`
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs

Comment (feedmeapples) [reply .github/workflows/release-base-image.yml:7]: cool

**Review by jbreiding — APPROVED** [review]
LGTM

**Review by alexshtin — APPROVED** [review]

**Pull request merged**